src/source/shift.rs

```diff
@@ -229,9 +229,10 @@ impl FromShift {
         JSON::Object(scope)
     }
 
-    fn dummy_parameter_scope<'a, I>(&self, params: I) -> json::JsonValue
+    fn parameter_scope_and_length<'a, I>(&self, params: I) -> (json::JsonValue, json::JsonValue)
```

Comment:
I'd be more comfortable if instead of a pair of `JsonValue` you had a `struct` with clearly-labeled names. Or maybe return a `usize` for the length and convert it to `json::JsonValue` at the call site?

Reply:
okay, I'll add struct with usize.